common/src/main/java/com/wynntils/services/mapdata/features/impl/MapAreaImpl.java

```diff
@@ -60,6 +62,12 @@ public List<Location> getPolygonArea() {
 
     @Override
     public BoundingPolygon getBoundingPolygon() {
-        return boundingPolygon;
+        return boundingPolygon != null
+                ? boundingPolygon
+                : (boundingPolygon = BoundingPolygon.fromLocations(polygonArea));
```

Comment:
Do you think it might be clearer to break this out into a calculateBoundingPolygon()? That way you can also call it directly in the constructor, and it would be so obvious that the comment would not be needed.